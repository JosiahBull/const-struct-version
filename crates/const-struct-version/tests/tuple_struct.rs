#![allow(dead_code)]

use const_struct_version::StructVersion;

#[derive(StructVersion)]
struct TupleStruct(u32, f32);

#[derive(StructVersion)]
struct TupleStruct2(f32, u32);

#[test]
fn test_tuple_struct_ordering_changes_hash() {
    let version = <TupleStruct as StructVersion>::version();
    let version2 = <TupleStruct2 as StructVersion>::version();

    insta::assert_debug_snapshot!(version);
    insta::assert_debug_snapshot!(version2);

    // Ensure all are different.
    assert_ne!(version, version2);
}
