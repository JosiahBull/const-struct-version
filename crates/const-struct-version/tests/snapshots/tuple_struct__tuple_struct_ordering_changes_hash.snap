---
source: crates/const-struct-version/tests/tuple_struct.rs
expression: version
---
"559af3f916a6a7b7b8282ee38df0f9fd930375da"
